### PR TITLE
bugfix: libtool: change url to download version 2.4.6 instead of 2.4.2

### PR DESCRIPTION
**Pull request by vishalkenchan**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libtool/package.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 class Libtool(Package):
     """libtool -- library building part of autotools"""
     homepage = "https://www.gnu.org/software/libtool/"
-    url      = "http://ftpmirror.gnu.org/libtool/libtool-2.4.2.tar.gz"
+    url      = "http://ftpmirror.gnu.org/libtool/libtool-2.4.6.tar.gz"
```
Comment (davydden) [inline var/spack/repos/builtin/packages/libtool/package.py]: the original version should work, AFAIK. So should happen is that the pattern is picked up from the `url` and then substituted using the currently chosen version. Does it not work for you without this change and `2.4.6`?

p.s. sometimes url parser fails, like for `tbb` and one has to give an url for each version https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/tbb/package.py#L14 . But here it looks easy and is supposed to work.

Comment (davydden) [inline var/spack/repos/builtin/packages/libtool/package.py]: ps. as an example, the basic url for trilinos is `12.2.1` https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/trilinos/package.py#L17, there is a whole range of versions up to `12.6.1`.

Comment (vishalkenchan) [inline var/spack/repos/builtin/packages/libtool/package.py]: Not sure why the url parser fails to pick up (requested) version 2.4.6 for me.

Comment (davydden) [inline var/spack/repos/builtin/packages/libtool/package.py]: what is the command-line output when you do `spack install libtool`?

Comment (vishalkenchan) [inline var/spack/repos/builtin/packages/libtool/package.py]: `spack uninstall libtool@2.4.6` and `spac uninstall libtool@2.4.2` and `spack install libtool` works without change to url to fetch 2.4.6 directly. Earlier it fetched 2.4.2 version for some reason. 
----
PS: With a fresh git clone of origin I got the same error as [issue 544](https://github.com/LLNL/spack/issues/544).

Comment (vishalkenchan) [inline var/spack/repos/builtin/packages/libtool/package.py]: `spack uninstall libtool@2.4.6` and `spac uninstall libtool@2.4.2` and `spack install libtool` works without changing the url to fetch 2.4.6 directly. Earlier it fetched 2.4.2 version for some reason.
ps: I had the same issue as [this](https://github.com/LLNL/spack/issues/544) with a fresh git clone.

**Pull request closed without merging**